test.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Error struct {
+	ErrorMessage string `json:"error"`
+}
+
+var link_map = map[string]string{}
+
+func getLinks(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("get link using post endpoint hit")
+
+	for key, value := range link_map {
+		fmt.Printf("[%s] = %s\n", key, value)
+	}
+	// fmt.Fprint(w, link_map)
+	jsonStr, err := json.Marshal(link_map)
+	if err != nil {
+		fmt.Printf("Error: %s", err.Error())
```

Comment:
In case of error user is not getting any response at least return error that something went wrong